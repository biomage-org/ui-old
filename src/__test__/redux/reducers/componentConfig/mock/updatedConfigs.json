[
  {
    "id": "embeddingCategoricalMain",
    "updatedConfig": {
      "axes": {
        "offset": 0,
        "gridWidth": 10,
        "xAxisText": "Umap 1",
        "yAxisText": "Umap 2",
        "domainWidth": 2,
        "gridOpacity": 0,
        "defaultValues": [
          "x",
          "y"
        ],
        "labelFontSize": 12,
        "titleFontSize": 13,
        "xAxisRotateLabels": false
      },
      "spec": "1.0.0",
      "title": {
        "dx": 10,
        "text": "",
        "anchor": "start",
        "fontSize": 20
      },
      "colour": {
        "invert": "standard",
        "gradient": "default",
        "masterColour": "#000000",
        "toggleInvert": "#FFFFFF",
        "reverseColourBar": false
      },
      "labels": {
        "size": 18,
        "enabled": false
      },
      "legend": {
        "colour": "#000000",
        "enabled": true,
        "position": "top"
      },
      "marker": {
        "size": 5,
        "shape": "circle",
        "opacity": 5,
        "showOpacity": true
      },
      "fontStyle": {
        "font": "sans-serif",
        "colour": "#000000"
      },
      "axesRanges": {
        "xMax": 10,
        "xMin": 0,
        "yMax": 10,
        "yMin": 0,
        "xAxisAuto": true,
        "yAxisAuto": true
      },
      "dimensions": {
        "width": 700,
        "height": 550
      }
    }
  },
  {
    "id": "frequencyPlotMain",
    "updatedConfig": {
      "axes": {
        "offset": 10,
        "gridWidth": 10,
        "xAxisText": "Sample",
        "yAxisText": "Proportion",
        "domainWidth": 2,
        "gridOpacity": 0,
        "labelFontSize": 12,
        "titleFontSize": 13,
        "xAxisRotateLabels": true
      },
      "spec": "1.0.0",
      "title": {
        "dx": 10,
        "text": "",
        "anchor": "start",
        "fontSize": 15
      },
      "colour": {
        "invert": "standard",
        "gradient": "default",
        "masterColour": "#000000",
        "toggleInvert": "#FFFFFF",
        "reverseColourBar": false
      },
      "labels": {
        "size": 18,
        "enabled": true
      },
      "legend": {
        "title": "Cell Set",
        "colour": "#000000",
        "offset": 40,
        "enabled": true,
        "position": "top"
      },
      "marker": {
        "size": 5,
        "shape": "circle",
        "opacity": 5,
        "showOpacity": true
      },
      "fontStyle": {
        "font": "sans-serif",
        "colour": "#000000"
      },
      "axesRanges": {
        "yMax": 10,
        "yMin": 0,
        "yAxisAuto": true
      },
      "dimensions": {
        "width": 500,
        "height": 500
      },
      "axisTitlesize": 13,
      "frequencyType": "proportional",
      "geneexpLegendloc": ""
    }
  },
  {
    "id": "trajectoryAnalysisMain",
    "updatedConfig": {
      "axes": {
        "offset": 0,
        "gridWidth": 10,
        "xAxisText": "Umap 1",
        "yAxisText": "Umap 2",
        "domainWidth": 2,
        "gridOpacity": 0,
        "defaultValues": [
          "x",
          "y"
        ],
        "labelFontSize": 12,
        "titleFontSize": 13,
        "xAxisRotateLabels": false
      },
      "spec": "1.0.0",
      "title": {
        "dx": 10,
        "text": "",
        "anchor": "start",
        "fontSize": 20
      },
      "colour": {
        "invert": "standard",
        "gradient": "default",
        "masterColour": "#000000",
        "toggleInvert": "#FFFFFF",
        "reverseColourBar": false
      },
      "labels": {
        "size": 18,
        "enabled": false
      },
      "legend": {
        "colour": "#000000",
        "enabled": true,
        "position": "top"
      },
      "marker": {
        "size": 20,
        "shape": "circle",
        "opacity": 5,
        "showOpacity": true
      },
      "fontStyle": {
        "font": "sans-serif",
        "colour": "#000000"
      },
      "axesRanges": {
        "xMax": 10,
        "xMin": 0,
        "yMax": 10,
        "yMin": 0,
        "xAxisAuto": true,
        "yAxisAuto": true
      },
      "dimensions": {
        "width": 700,
        "height": 550
      },
      "selectedSample": "All",
      "selectedCellSet": "louvain"
    }
  },
  {
    "id": "ViolinMain",
    "updatedConfig": {
      "axes": {
        "offset": 10,
        "gridWidth": 10,
        "xAxisText": "",
        "yAxisText": "",
        "domainWidth": 2,
        "gridOpacity": 0,
        "labelFontSize": 12,
        "titleFontSize": 13,
        "xAxisRotateLabels": true
      },
      "spec": "1.0.0",
      "title": {
        "dx": 0,
        "text": "Gzma",
        "anchor": "start",
        "fontSize": 20
      },
      "colour": {
        "invert": "standard",
        "gradient": "default",
        "masterColour": "#000000",
        "toggleInvert": "#FFFFFF",
        "reverseColourBar": false
      },
      "labels": {
        "size": 18,
        "enabled": true
      },
      "legend": {
        "colour": "#000000",
        "enabled": false,
        "position": "top"
      },
      "marker": {
        "size": 5,
        "shape": "circle",
        "opacity": 5,
        "showOpacity": true
      },
      "fontStyle": {
        "font": "sans-serif",
        "colour": "#000000"
      },
      "shownGene": "Gzma",
      "axesRanges": {
        "yMax": 10,
        "yMin": 0,
        "yAxisAuto": true
      },
      "dimensions": {
        "width": 550,
        "height": 400
      },
      "normalised": "normalised",
      "kdeBandwidth": 0.3,
      "keepValuesOnReset": [
        "shownGene",
        "title"
      ],
      "statisticsVisible": false,
      "selectedPointsVisible": true
    }
  },
  {
    "id": "ViolinMain-0",
    "updatedConfig": {
      "axes": {
        "offset": 10,
        "gridWidth": 10,
        "xAxisText": "",
        "yAxisText": "",
        "domainWidth": 2,
        "gridOpacity": 0,
        "labelFontSize": 12,
        "titleFontSize": 13,
        "xAxisRotateLabels": true
      },
      "spec": "1.0.0",
      "title": {
        "dx": 0,
        "text": "Lyz2",
        "anchor": "start",
        "fontSize": 20
      },
      "colour": {
        "invert": "standard",
        "gradient": "default",
        "masterColour": "#000000",
        "toggleInvert": "#FFFFFF",
        "reverseColourBar": false
      },
      "labels": {
        "size": 18,
        "enabled": true
      },
      "legend": {
        "colour": "#000000",
        "enabled": false,
        "position": "top"
      },
      "marker": {
        "size": 5,
        "shape": "circle",
        "opacity": 5,
        "showOpacity": true
      },
      "fontStyle": {
        "font": "sans-serif",
        "colour": "#000000"
      },
      "shownGene": "Lyz2",
      "axesRanges": {
        "yMax": 10,
        "yMin": 0,
        "yAxisAuto": true
      },
      "dimensions": {
        "width": 550,
        "height": 400
      },
      "normalised": "normalised",
      "kdeBandwidth": 0.3,
      "keepValuesOnReset": [
        "shownGene",
        "title"
      ],
      "statisticsVisible": false,
      "selectedPointsVisible": true
    }
  }
]