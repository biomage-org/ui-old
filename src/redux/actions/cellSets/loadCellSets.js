import fetchAPI from 'utils/http/fetchAPI';
import handleError from 'utils/http/handleError';
import {
  CELL_SETS_LOADED, CELL_SETS_LOADING, CELL_SETS_ERROR,
} from 'redux/actionTypes/cellSets';
import endUserMessages from 'utils/endUserMessages';

import config from 'config';
import { api } from 'utils/constants';

const loadCellSets = (experimentId, forceReload = false) => async (dispatch, getState) => {
  const {
    loading, error, updatingClustering,
  } = getState().cellSets;

  if (!forceReload && ((!loading && !error) || updatingClustering)) {
    return null;
  }

  if (error) {
    dispatch({
      type: CELL_SETS_LOADING,
    });
  }

  let url;

  if (config.currentApiVersion === api.V1) {
    url = `/v1/experiments/${experimentId}/cellSets`;
  } else if (config.currentApiVersion === api.V2) {
    url = `/v2/experiments/${experimentId}/cellSets`;
  }

  try {
    const data = await fetchAPI(url);

    dispatch({
      type: CELL_SETS_LOADED,
      payload: {
        experimentId,
        data: data.cellSets,
      },
    });
  } catch (e) {
    const errorMessage = handleError(e, endUserMessages.ERROR_FETCHING_CELL_SETS);

    dispatch({
      type: CELL_SETS_ERROR,
      payload: { error: errorMessage },
    });
  }
};

export default loadCellSets;
