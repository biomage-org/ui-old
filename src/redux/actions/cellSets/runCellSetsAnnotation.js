import {
  CELL_SETS_ERROR, CELL_SETS_CLUSTERING_UPDATING,
} from 'redux/actionTypes/cellSets';

import getTimeoutForWorkerTask from 'utils/getTimeoutForWorkerTask';
import fetchWork from 'utils/work/fetchWork';

const runCellSetsAnnotation = (experimentId, species, tissue) => async (dispatch, getState) => {
  const { error, updatingClustering, loading } = getState().cellSets;

  if ((loading && updatingClustering) || error) return;

  const body = {
    name: 'ScTypeAnnotate',
    species,
    tissue,
  };

  dispatch({
    type: CELL_SETS_CLUSTERING_UPDATING,
  });

  const timeout = getTimeoutForWorkerTask(getState(), 'ClusterCells');

  try {
    await fetchWork(
      experimentId,
      body,
      getState,
      dispatch,
      {
        timeout,
        broadcast: true,
      },
    );
  } catch (e) {
    dispatch({
      type: CELL_SETS_ERROR,
      payload: {
        experimentId,
        error: e,
      },
    });
  }
};

export default runCellSetsAnnotation;
