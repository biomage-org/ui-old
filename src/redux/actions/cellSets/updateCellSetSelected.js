import { CELL_SETS_SET_SELECTED } from '../../actionTypes/cellSets';

const updateCellSetSelected = (keys) => async (dispatch, getState) => {
  const {
    loading, error,
  } = getState().cellSets;

  if (loading || error) {
    return null;
  }

  await dispatch({
    type: CELL_SETS_SET_SELECTED,
    payload: {
      keys,
    },
  });
};

export default updateCellSetSelected;
