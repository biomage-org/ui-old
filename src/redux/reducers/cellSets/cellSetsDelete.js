/* eslint-disable no-param-reassign */
import _ from 'lodash';
import produce, { current } from 'immer';

const cellSetsDelete = produce((draft, action) => {
  const { key } = action.payload;
  const { parentNodeKey } = current(draft).properties[key];

  // Remove cellSet from hierarchy
  const cellClassIndex = draft.hierarchy.findIndex((rootNode) => rootNode.key === parentNodeKey);
  _.remove(draft.hierarchy[cellClassIndex].children, ({ key: currentKey }) => currentKey === key);

  // Delete from the properties as well.
  delete draft.properties[key];

  // If the key was in the list of selected keys, make sure we remove it from there.
  _.remove(draft.selected, (currentSelectedKey) => currentSelectedKey === key);

  // Delete from hidden if it was selected to be hidden.
  draft.hidden.delete(key);
});

export default cellSetsDelete;
