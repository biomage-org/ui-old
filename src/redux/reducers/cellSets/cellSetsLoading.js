/* eslint-disable no-param-reassign */
import produce from 'immer';

import initialState from 'redux/reducers/cellSets/initialState';

const cellSetsLoading = produce((draft) => {
  draft.initialLoadPending = false;
  draft.loading = true;
  draft.error = false;
}, initialState);

export default cellSetsLoading;
