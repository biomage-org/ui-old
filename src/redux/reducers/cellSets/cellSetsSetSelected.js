/* eslint-disable no-param-reassign */
import produce from 'immer';

const cellSetsSetSelected = produce((draft, action) => {
  const { keys } = action.payload;

  draft.selected = keys;
});

export default cellSetsSetSelected;
