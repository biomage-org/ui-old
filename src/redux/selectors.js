/* eslint-disable import/prefer-default-export */

// Accumulates all the different selectors

import * as backendSelectors from './selectors/backendStatus';
import * as cellSetsSelectors from './selectors/cellSets';
import * as componentConfigSelectors from './selectors/componentConfig';
import * as genesSelectors from './selectors/genes';

const getBackendStatus = (...params) => (state) => (
  backendSelectors.getBackendStatus(...params)(state.backendStatus));

const getCellSets = (...params) => (state) => (
  cellSetsSelectors.getCellSets(...params)(state.cellSets));

const getCellSetsHierarchy = (...params) => (state) => (
  cellSetsSelectors.getCellSetsHierarchy(...params)(state.cellSets));

const getCellSetsHierarchyByType = (...params) => (state) => (
  cellSetsSelectors.getCellSetsHierarchyByType(...params)(state.cellSets));

const getCellSetsHierarchyByKeys = (...params) => (state) => (
  cellSetsSelectors.getCellSetsHierarchyByKeys(...params)(state.cellSets));

const getPlotConfigs = (...params) => (state) => (
  componentConfigSelectors.getPlotConfigs(...params)(state.componentConfig));

const getSavedPlots = (...params) => (state) => (
  componentConfigSelectors.getSavedPlots(...params)(state.componentConfig));

const getGeneList = (...params) => (state) => (
  genesSelectors.getGeneList(...params)(state.genes));

export {
  getBackendStatus,
  getCellSets,
  getCellSetsHierarchy,
  getCellSetsHierarchyByType,
  getCellSetsHierarchyByKeys,
  getPlotConfigs,
  getSavedPlots,
  getGeneList,
};
