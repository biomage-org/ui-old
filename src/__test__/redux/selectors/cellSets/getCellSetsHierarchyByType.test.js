import { getCellSetsHierarchyByType } from 'redux/selectors';
import { mockCellSets } from '__test__/test-utils/cellSets.mock';

describe('Get cell sets hierarchy selector by type test', () => {
  it('returns correctly by one key', () => {
    const returnedHierarchy = getCellSetsHierarchyByType('firstType')({ cellSets: { accessible: true, ...mockCellSets } });
    expect(returnedHierarchy).toMatchSnapshot();
  });
});
