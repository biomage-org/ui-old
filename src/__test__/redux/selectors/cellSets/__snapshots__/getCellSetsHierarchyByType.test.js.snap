// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get cell sets hierarchy selector by type test returns correctly by one key 1`] = `[]`;
